samples/msal-node-samples/EndToEndTestApp/app/test/testUtils.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Screenshot } from "../../../../e2eTestUtils/TestUtils";
+import { Page } from "puppeteer";
+import fs from "fs";
+
+// Constants
+export const SCREENSHOT_BASE_FOLDER_NAME = `${__dirname}/screenshots`;
+export const SAMPLE_HOME_URL = "http://localhost:3000";
+export const SUCCESSFUL_GRAPH_CALL_ID = "graph-called-successfully";
+export const SUCCESSFUL_GET_ALL_ACCOUNTS_ID = "accounts-retrieved-successfully";
+
+export async function enterCredentials(page: Page, screenshot: Screenshot, username: string, accountPwd: string): Promise<void> {
+    await page.waitForSelector("#i0116");
+    await screenshot.takeScreenshot(page, "loginPage");
+    await page.type("#i0116", username);
+    await screenshot.takeScreenshot(page, "loginPageUsernameFilled")
+    await page.click("#idSIButton9");
+    await page.waitForSelector("#idA_PWD_ForgotPassword");
+    await page.waitForSelector("#idSIButton9");
+    await screenshot.takeScreenshot(page, "pwdInputPage");
+    await page.type("#i0118", accountPwd);
+    await screenshot.takeScreenshot(page, "loginPagePasswordFilled")
+    await page.click("#idSIButton9");
+}
+
+export async function enterCredentialsWithConsent(page: Page, screenshot: Screenshot, username: string, accountPwd: string): Promise<void> {
+    await this.enterCredentials(page, screenshot, username, accountPwd);
+    await this.approveConsent(page, screenshot);
+}
+
+export async function enterCredentialsADFSWithConsent(page: Page, screenshot: Screenshot, username: string, accountPwd: string): Promise<void> {
+    await this.enterCredentialsADFS(page, screenshot, username, accountPwd);
+    await this.approveConsent(page, screenshot);
+}
+
+export async function approveConsent(page: Page, screenshot: Screenshot): Promise<void> {
+    await page.waitForSelector("#idSIButton9");
+    await page.click("#idSIButton9");
+    await takeScreenshotAfter(6000, screenshot, page, 'consentApproved'); 
```

Comment:
6 seconds is a bit long to wait just for a screenshot, is it not? Let's see if we can find something better to wait for than a fixed time

Reply:
Reduced this timeout as well.